EESSI-extend-2023.06-easybuild.eb

```diff
@@ -0,0 +1,96 @@
+ (EESSI, pronounced as "easy") is to build a common stack of scientific
+ software installations for HPC systems and beyond, including laptops,
+ personal workstations and cloud infrastructure.
+
+ This module allows you to extend EESSI using the same configuration for
+ EasyBuild as EESSI itself uses. The default installation is the users
+ home directory, but this can be overridden for a site installation (by setting
+ EESSI_SITE_INSTALL) or for a direct installation under CVMFS (by setting
+ EESSI_CVMFS_INSTALL). 
+"""
+
+toolchain = SYSTEM
+
+source_urls = None
+sources = ['eb_hooks.py']
+checksums = ['8ae609f99b6953beae89aa1945913686c86c156509dbc55e2b6b8017b13fcf66']
+
+# Don't extend the PATH
+prepend_to_path = None
+
+# All the dependencies we filter in EESSI
+local_deps_to_filter = "Autoconf,Automake,Autotools,binutils,bzip2,DBus,flex,gettext,gperf,help2man,intltool,libreadline,libtool,M4,makeinfo,ncurses,util-linux,XZ,zlib"
+local_arch_specific_deps_to_filter = {'aarch64': ',yasm', 'x86_64': ''}
+local_deps_to_filter += local_arch_specific_deps_to_filter[ARCH]
+
+modextravars = {
+    'EASYBUILD_FILTER_DEPS': local_deps_to_filter,
+    'EASYBUILD_IGNORE_OSDEPS': '1',
+    'EASYBUILD_DEBUG': '1',
+    'EASYBUILD_TRACE': '1',
+    'EASYBUILD_ZIP_LOGS': 'bzip2',
+    'EASYBUILD_RPATH': '1',
+    'EASYBUILD_FILTER_ENV_VARS': 'LD_LIBRARY_PATH',
+    'EASYBUILD_READ_ONLY_INSTALLDIR': '1',
+    'EASYBUILD_MODULE_EXTENSIONS': '1',
+    'EASYBUILD_EXPERIMENTAL': '1',
+    'EASYBUILD_HOOKS': '%(installdir)s/eb_hooks.py',
+}
+
+# Need a few other variables, but they are more dynamic
+# EASYBUILD_SYSROOT=${EPREFIX}
+# EASYBUILD_PREFIX=${WORKDIR}/easybuild
+# EASYBUILD_INSTALLPATH=${EESSI_PREFIX}/software/${EESSI_OS_TYPE}/${EESSI_SOFTWARE_SUBDIR}
+# EASYBUILD_SOURCEPATH=${WORKDIR}/easybuild/sources:${EESSI_SOURCEPATH}
+modluafooter = """
+if (mode() == "load") then
+  -- Use a working directory for temporary build files
+  if (os.getenv("WORKING_DIR") == nil) then
+    LmodMessage("-- Using /tmp/$USER as a temporary working directory for installations, you can override this by setting the environment variable WORKING_DIR and reloading the module (e.g., /dev/shm is a common option)")
+  end
+end
+working_dir = os.getenv("WORKING_DIR") or pathJoin("/tmp", os.getenv("USER"))
+-- Gather the EPREFIX to use as a sysroot
+sysroot = os.getenv("EESSI_EPREFIX")
+-- Use an installation prefix that we _should_ have write access to
+if (os.getenv("EESSI_CVMFS_INSTALL") ~= nil) then
+  easybuild_installpath = os.getenv("EESSI_SOFTWARE_PATH")
+elseif (os.getenv("EESSI_SITE_INSTALL") ~= nil) then 
+  easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), 'versions', 'host_injections')
+else
+  -- Would have liked to use os.getenv("EESSI_CVMFS_REPO") here but the hypen needs to be escaped
+  easybuild_installpath = string.gsub(os.getenv("EESSI_SOFTWARE_PATH"), "/cvmfs/pilot.eessi%-hpc.org", pathJoin(os.getenv("HOME"), "eessi"))
```

Comment:
I see a future where this could benefit from the same thing we do for EasyBuild, i.e. going through a list of env vars to use as the base, i.e. HOME, PROJECT, etc.

Reply:
The benefit of the current setup is that we can configure EESSI to automatically expose the modules. Thinking about it though, it may be more interesting to do this in a more dynamic way inside the module itself...but the downside of this is  you would have to always load the module to see the installed modules.